### DISCUSSION TITLE
State of the Code

### DISCUSSION THREAD

**Discussion by aersam** [dotnet/SqlClient#1605] [General]
It looks a bit like there is some agreement that the code in this repo could be improved: https://github.com/dotnet/datalab/issues/6 

I personally think resources should go into a new implementation here. The speed new features arrives is not satisfying to me and quite basic issues like #1108 or #1009 seem to be very hard to do - not possible for a community member. Also the compat with the old .Net Framework driver seems to complicate stuff. 

Just wanted to open a discussion here, hope it's helpful. 

That being said I very much appreciate the development of this driver and the features and bug fixes it gets, I just think there should be a a good discussion about the future of it

**Comment by David-Engel** [comment]
I agree that big changes are difficult in this library. There is a lot of good discussion in that datalab thread that talks about the reasons it is the way it is.

Any 20+ year old code base that has had dozens of different people contributing to it will need improvement. There are too many customers and existing implementations to put this library on hold to start a new implementation. It still needs changes to support new SQL Server features that customers expect to be available in the client. A new implementation will need to be funded in addition to the budget that funds development on this library.

Backwards compatibility and stability are difficult and expensive to maintain for decades but they have always been extremely important to customers.
